File: spec/lua/test_eos_exception_u4.lua
local luaunit = require("luaunit")

require("eos_exception_u4")

TestEosExceptionU4 = {}

function TestEosExceptionU4:test_eos_exception_u4()
    luaunit.assertErrorMsgMatches(".+: requested %d+ bytes, but got only %d+ bytes", EosExceptionU4.from_file, EosExceptionU4, "src/term_strz.bin")
end

File: spec/python/test_eos_exception_u4.py
from base_test import BaseTest
from eos_exception_u4 import EosExceptionU4

class TestEosExceptionU4(BaseTest):
    def test_eos_exception_u4(self):
        with self.assertRaisesRegex(EOFError, r"^requested \d+ bytes, but only \d+ bytes available$"):
            with EosExceptionU4.from_file('src/term_strz.bin') as r:
                pass
